### PR TITLE
Charging for deploy 4.8

### PR DESCRIPTION
**Pull request by ArturGajowy**
## Overview
<sup>_What this PR does, and why it's needed_</sup>

Yet another set of refactors and cleanups needed for charging for deploy to be introduced safely.

### JIRA ticket:
<sup>_Create it if there isn't one already._</sup>
https://rchain.atlassian.net/browse/RCHAIN-3135

### Notes
<sup>_Optional. Add any notes on caveats, approaches you tried that didn't work, or anything else._</sup>

### Please make sure that this PR:
- [x] is at most 200 lines of code (excluding tests),
- [x] meets [RChain development coding standards](https://rchain.atlassian.net/wiki/spaces/DOC/pages/28082177/Coding+Standards),
- [x] includes tests for all added features,
- [x] has a reviewer assigned,
- [x] has [all commits signed](https://rchain.atlassian.net/wiki/spaces/DOC/pages/498630673/How+to+sign+commits+to+rchain+rchain).

### [Bors](https://bors.tech/) cheat-sheet:

- `bors r+` runs integration tests and merges the PR (if it's approved),
- `bors try` runs integration tests for the PR,
- `bors delegate+` enables non-maintainer PR authors to run the above.

### CODE REVIEW

File: casper/src/test/scala/coop/rchain/casper/MultiParentCasperRholangSpec.scala [hunk 1]
```diff
@@ -53,8 +53,7 @@ class MultiParentCasperRholangSpec extends FlatSpec with Matchers with Inspector
   }
 
   it should "be able to use the registry" in effectTest {
-    HashSetCasperTestNode.standaloneEff(genesis, validatorKeys.head).use { implicit node =>
-      import node.casperEff
+    HashSetCasperTestNode.standaloneEff(genesis, validatorKeys.head).use { node =>
       implicit val rm = node.runtimeManager
 
       val registerDeploy = ConstructDeploy.sourceDeploy(
```
Comment (JosDenmark) [inline casper/src/test/scala/coop/rchain/casper/MultiParentCasperRholangSpec.scala]: A simple alternative to fixing a timestamp is to just use `RegistrySigGen` to calculate the URI.

File: casper/src/test/scala/coop/rchain/casper/MultiParentCasperAddBlockSpec.scala [hunk 2]
```diff
@@ -95,26 +96,26 @@ class MultiParentCasperAddBlockSpec extends FlatSpec with Matchers with Inspecto
...
+                    .sourceDeployNowF("contract @\"add\"(@x, @y, ret) = { ret!(x + y) }")
+        signedBlock1 <- node.addBlock(deploy1)
+        deploy2 <- ConstructDeploy
+                    .sourceDeployNowF("new unforgable in { @\"add\"!(5, 7, *unforgable) }")
+        signedBlock2 <- node.addBlock(deploy2)
         _            = ProtoUtil.parentHashes(signedBlock2) should be(Seq(signedBlock1.blockHash))
         dag          <- node.casperEff.blockDag
         estimate     <- node.casperEff.estimator(dag)
         _            = estimate shouldBe IndexedSeq(signedBlock2.blockHash)
         // channel is deterministic because of the fixed timestamp
```
Comment (JosDenmark) [inline casper/src/test/scala/coop/rchain/casper/MultiParentCasperAddBlockSpec.scala]: This comment can be removed now.

**Comment by JosDenmark** [comment]
LGTM

File: casper/src/main/scala/coop/rchain/casper/util/rholang/RuntimeManager.scala [hunk 3]
```diff
@@ -48,7 +48,8 @@ trait RuntimeManager[F[_]] {
   def replayComputeState(startHash: StateHash)(
       terms: Seq[InternalProcessedDeploy],
       blockData: BlockData,
-      invalidBlocks: Map[BlockHash, Validator]
+      invalidBlocks: Map[BlockHash, Validator],
+      isGenesis: Boolean
```
Comment (tymm) [inline casper/src/main/scala/coop/rchain/casper/util/rholang/RuntimeManager.scala]: Quick explanation in the commit message why we have to pass along this information would be helpful.

File: casper/src/test/scala/coop/rchain/casper/MultiParentCasperRholangSpec.scala (hunk 1 above)
Comment (ArturGajowy) [reply casper/src/test/scala/coop/rchain/casper/MultiParentCasperRholangSpec.scala]: Will address that in an even more elegant manner after we have `deployId` merged

File: casper/src/main/scala/coop/rchain/casper/util/rholang/RuntimeManager.scala (hunk 3 above)
Comment (ArturGajowy) [reply casper/src/main/scala/coop/rchain/casper/util/rholang/RuntimeManager.scala]: Done

**Comment by ArturGajowy** [comment]
> Thanks for adding dynamic unforgeable names with RegistrySigGen.generateUnforgeableNameId

Kudos go to @JosDenmark :)

**Pull request merged**